local/modules/Front/Controller/CustomerController.php

```diff
@@ -331,8 +332,15 @@ public function updateAction()
 
                 $customerChangeEvent = $this->createEventInstance($form->getData());
                 $customerChangeEvent->setCustomer($customer);
-                // We do not allow customer email modification
-                $customerChangeEvent->setEmailUpdateAllowed(false);
+
+                $configCustomerChangeEmail = intval(ConfigQuery::read('customer_change_email', 0));
+
+                if ($configCustomerChangeEmail) {

```

Comment:
you can directly test the value returned by ConfigQuery::read here, it is not needed to put it in an intermediary variable